java-checks/src/main/resources/org/sonar/l10n/java/rules/java/S5659_java.html

```diff
@@ -1,98 +1,58 @@
 </ul>
-</div>
-<div class="paragraph">
-<p>In addition, be extra careful when using <a href="https://github.com/jwtk/jjwt">jwtk/Java JWT</a> library "<code>parse</code>" method, parsing a signed token (JWT + JWS (signature)) or an unsigned one. To guess how to parse the token the parse method will look at the token headers (which are not signed and thus could be forged by an attacker). This attack is sometimes referred as the "None algorithm attack". Instead, you should consider using "<code>parseClaimsJws</code>" parsing signed token. If the signature is not provided together with the JWT, the method will fail as expected.</p>
-</div>
-<div class="sect1">
-<h2 id="_noncompliant_code_examples">Noncompliant Code Examples</h2>
-<div class="sectionbody">
-<div class="paragraph">
-<p>Using <a href="https://github.com/jwtk/jjwt">jwtk/Java JWT</a> library:</p>
-</div>
-<div class="listingblock">
-<div class="content">
-<pre>// Signinig:
+<h2>Noncompliant Code Example</h2>
+<p>Using <a href="https://github.com/jwtk/jjwt">jwtk/Java JWT</a> library (to verify a signed token (containing a JWS) don't use the
+<code>parse</code> method as it doesn't throw an exception if an unsigned token is provided):</p>
+<pre>
+// Signing:
 io.jsonwebtoken.Jwts.builder() // Noncompliant, token is not signed.
   .setSubject(USER_LOGIN)
   .compact();
 // Verifying:
-io.jsonwebtoken.Jwts.parser().setSigningKey(SECRET_KEY).parse(token).getBody(); // Noncompliant, if the token has no signature, this method will still parse it correctly.</pre>
-</div>
-</div>
-<div class="paragraph">
+io.jsonwebtoken.Jwts.parser().setSigningKey(SECRET_KEY).parse(token).getBody(); // Noncompliant
+</pre>
 <p>Using <a href="https://github.com/auth0/java-jwt">auth0/Java JWT</a> library:</p>
-</div>
-<div class="listingblock">
-<div class="content">
-<pre>// Signinig:
+<pre>
+// Signing:
 com.auth0.jwt.JWT.create()
   .withSubject(SUBJECT)
   .sign(Algorithm.none()); // Noncompliant, use only strong cipher algorithms when signing this JWT.
 // Verifying:
 JWTVerifier nonCompliantVerifier = com.auth0.jwt.JWT.require(Algorithm.none()) // Noncompliant
   .withSubject(LOGIN)
-  .build();</pre>
-</div>
-</div>
-</div>
-</div>
-<div class="sect1">
-<h2 id="_compliant_solution">Compliant Solution</h2>
-<div class="sectionbody">
-<div class="paragraph">
-<p>Using <a href="https://github.com/jwtk/jjwt">Java JWT</a> library:</p>
-</div>
-<div class="listingblock">
-<div class="content">
-<pre>// Signinig:
+  .build();
+</pre>
+<h2>Compliant Solution</h2>
+<p>Using <a href="https://github.com/jwtk/jjwt">Java JWT</a> library (to verify a signed token (containing a JWS) use the <code>parseClaimsJws</code>
+method that will throw an exception if an unsigned token is provided):</p>
+<pre>
+// Signing:
 Jwts.builder() // Compliant
   .setSubject(USER_LOGIN)
-  .signWith(SignatureAlgorithm.H{rule:java:S256}, SECRET_KEY)
+  .signWith(SignatureAlgorithm.HS256, SECRET_KEY)
```

Comment:
For the record, the changes in this file are due to the fact that the new "gh_generate" was used to generate this file. As we can see in this line, this new way still contains some bugs, we should keep using the old one for now.